client/www/scripts/modules/manager/manager.controllers.js

```diff
@@ -609,6 +616,59 @@ Manager.controller('ManagerMainController', [
       }
     };
 
+    function updateHostEnv(host, env) {
+      var deferred = $q.defer();
+      var command =  {
+        cmd: 'env-set',
+        env: env
+      };
+
+      if (host) {
+        host.action(command, function(err, res) {
+          if (err) {
+            deferred.reject(err);
+          } else {
+            deferred.resolve(env);
+          }
+        });
+      }
```

Comment:
consider re-writing the above method  as we have a different convention for async/promises than $q.defer()
``
function updateHostEnv(host, env) {
  if ( !host) {
    return;
  }
  var command - {};
  return host.action(command, function(err, res) {
    if (err) {
      $log.warn('bad...' + err.message;
      return;
    }
    return env;
  }
}
``